src/mscorlib/src/System/Runtime/Intrinsics/X86/Sse2.cs

```diff
@@ -1023,47 +1024,96 @@ public static class Sse2
+            return Sse.StaticCast<uint, byte>(Shuffle(
+                    Sse.StaticCast<ushort, uint>(UnpackLow(vector2, vector2)), 0));
+        }
         /// <summary>
         /// __m128i _mm_set1_epi8 (char a)
         ///   HELPER
         /// </summary>
-        public static Vector128<sbyte> SetAllVector128(sbyte value) => SetAllVector128(value);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<sbyte> SetAllVector128(sbyte value)
+        {
+            Vector128<sbyte> vector = Sse.StaticCast<int, sbyte>(ConvertScalarToVector128Int32(value));
+            Vector128<short> vector2 = Sse.StaticCast<sbyte, short>(UnpackLow(vector, vector));
+            return Sse.StaticCast<int, sbyte>(Shuffle(
+                Sse.StaticCast<short, int>(UnpackLow(vector2, vector2)), 0));
+        }
         /// <summary>
         /// __m128i _mm_set1_epi16 (short a)
         ///   HELPER
         /// </summary>
-        public static Vector128<short> SetAllVector128(short value) => SetAllVector128(value);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<short> SetAllVector128(short value)
+        {
+            Vector128<short> vector = Sse.StaticCast<int, short>(ConvertScalarToVector128Int32(value));
+            return Sse.StaticCast<int, short>(Shuffle(
+                Sse.StaticCast<short, int>(UnpackLow(vector, vector)), (byte)0));
+        }
         /// <summary>
         /// __m128i _mm_set1_epi16 (short a)
         ///   HELPER
         /// </summary>
-        public static Vector128<ushort> SetAllVector128(ushort value) => SetAllVector128(value);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<ushort> SetAllVector128(ushort value)
+        {
+            Vector128<ushort> vector = Sse.StaticCast<uint, ushort>(ConvertScalarToVector128UInt32(value));
+            return Sse.StaticCast<uint, ushort>(Shuffle(
+                Sse.StaticCast<ushort,uint>(UnpackLow(vector, vector)), (byte)0));
+        }
         /// <summary>
         /// __m128i _mm_set1_epi32 (int a)
         ///   HELPER
         /// </summary>
-        public static Vector128<int> SetAllVector128(int value) => SetAllVector128(value);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<int> SetAllVector128(int value)
+        {
+            return Shuffle(ConvertScalarToVector128Int32(value), 0);
+        }
         /// <summary>
         /// __m128i _mm_set1_epi32 (int a)
         ///   HELPER
         /// </summary>
-        public static Vector128<uint> SetAllVector128(uint value) => SetAllVector128(value);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<uint> SetAllVector128(uint value)
+        {
+            return Shuffle(ConvertScalarToVector128UInt32(value), 0);
```

Comment:
could you break these lines apart, with temps and comments.

The JIT should be able to properly optimize them, and it makes it easier to read for us.

Reply:
Will do